base/server/python/pki/server/deployment/__init__.py

```diff
@@ -228,6 +228,68 @@ def flatten_master_dict(self):
             subsystem_dict[0] = None
             self.mdict.update(subsystem_dict)
 
+    def configure_id_generators(self, subsystem):
+
+        if subsystem.type == 'CA':
+
+            request_id_generator = self.mdict['pki_request_id_generator']
+
+            if request_id_generator == 'random':
+                subsystem.config['dbs.request.id.generator'] = request_id_generator
+                subsystem.config['dbs.request.id.length'] = self.mdict['pki_request_id_length']
+
+            else:  # legacy
+                subsystem.config['dbs.beginRequestNumber'] = '1'
+                subsystem.config['dbs.endRequestNumber'] = '10000000'
+                subsystem.config['dbs.requestIncrement'] = '10000000'
+                subsystem.config['dbs.requestLowWaterMark'] = '2000000'
+                subsystem.config['dbs.requestCloneTransferNumber'] = '10000'
+                subsystem.config['dbs.requestRangeDN'] = 'ou=requests,ou=ranges'
+
+            cert_id_generator = self.mdict['pki_cert_id_generator']
+
+            if cert_id_generator == 'random':
+                subsystem.config['dbs.cert.id.generator'] = cert_id_generator
+                subsystem.config['dbs.cert.id.length'] = self.mdict['pki_cert_id_length']
+
+            else:  # legacy
+                subsystem.config['dbs.beginSerialNumber'] = '1'
+                subsystem.config['dbs.endSerialNumber'] = '10000000'
+                subsystem.config['dbs.serialIncrement'] = '10000000'
+                subsystem.config['dbs.serialLowWaterMark'] = '2000000'
+                subsystem.config['dbs.serialCloneTransferNumber'] = '10000'
+                subsystem.config['dbs.serialRangeDN'] = 'ou=certificateRepository,ou=ranges'
+
+        if subsystem.type == 'KRA':
+
+            request_id_generator = self.mdict['pki_key_id_generator']
```

Comment:
I think this should be equivalent to the CA case - the original config values are identical?
`request_id_generator = self.mdict['pki_request_id_generator']`

Reply:
Good catch!